File: java-hello-world/01-hfs-files/src/main/java/hfsfiles/ScriptHfsFiles.java
package hfsfiles;

import com.hedera.hashgraph.sdk.*;
import io.github.cdimascio.dotenv.Dotenv;

import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

public class ScriptHfsFiles {
    public static void main(String[] args) throws Exception {
        // Ensure required environment variables are available
        Dotenv dotenv = Dotenv.load();
        String accountIdStr = dotenv.get("ACCOUNT_ID");
        String privateKeyStr = dotenv.get("ACCOUNT_PRIVATE_KEY");
        if (accountIdStr == null || privateKeyStr == null) {
            throw new RuntimeException("Please set required keys in .env file.");
        }
        if (privateKeyStr.startsWith("0x")) {
            privateKeyStr = privateKeyStr.substring(2);
        }

        // Configure client using environment variables
        AccountId accountId = AccountId.fromString(accountIdStr);
        PrivateKey accountKey = PrivateKey.fromStringECDSA(privateKeyStr);
        Client client = Client.forTestnet().setOperator(accountId, accountKey);

        // Read file from disk
        String localFileContents = new String(Files.readAllBytes(Paths.get("my-file.txt")), StandardCharsets.UTF_8);

        // Write file onto Hedera Testnet, using HFS FileCreateTransaction
        FileCreateTransaction fileCreateTx = new FileCreateTransaction()
            // NOTE: File create transaction
            // Step (1) in the accompanying tutorial
            // /* ... */
            .setContents(localFileContents)
            .freezeWith(client);
        FileCreateTransaction fileCreateTxSigned = fileCreateTx.sign(accountKey);
        TransactionResponse fileCreateTxSubmitted = fileCreateTxSigned.execute(client);
        TransactionId fileCreateTxId = fileCreateTxSubmitted.transactionId;
        TransactionReceipt fileCreateTxReceipt = fileCreateTxSubmitted
            .setValidateStatus(true)
            .getReceipt(client);
        FileId fileId = fileCreateTxReceipt.fileId;

        // Read file from Hedera Testnet, using HFS FileContentsQuery
        FileContentsQuery fileReadQuery = new FileContentsQuery()
            // NOTE: File contents query
            // Step (2) in the accompanying tutorial
            // /* ... */;
            .setFileId(fileId);
        String networkFileContents = fileReadQuery.execute(client).toStringUtf8();

        // Output results
        String txExplorerUrl = "https://hashscan.io/testnet/transaction/" + fileCreateTxId;
        System.out.println("fileId: " + fileId);
        System.out.println("fileCreateTxId: " + fileCreateTxId);
        System.out.println("txExplorerUrl: " + txExplorerUrl);
        System.out.println("localFileContents: " + localFileContents);
        System.out.println("networkFileContents: " + networkFileContents);

        client.close();
    }
}

File: golang-hello-world/01-hfs-files/script-hfs-files.go
package main

import (
	"fmt"
	"log"
	"os"
	"strings"

	"github.com/hashgraph/hedera-sdk-go/v2"
	"github.com/joho/godotenv"
)

func main() {
	// Load environment variables from .env file
	err := godotenv.Load()
	if err != nil {
		log.Fatal("Error loading .env")
	}

	accountIdStr := os.Getenv("ACCOUNT_ID")
	accountKeyStr := os.Getenv("ACCOUNT_PRIVATE_KEY")
	accountKeyStr = strings.TrimPrefix(accountKeyStr, "0x")

	accountId, err := hedera.AccountIDFromString(accountIdStr)
	if err != nil {
		log.Fatalf("Error parsing account ID: %v", err)
	}
	accountKey, err := hedera.PrivateKeyFromStringECDSA(accountKeyStr)
	if err != nil {
		log.Fatalf("Error parsing account private key: %v", err)
	}
	client := hedera.ClientForTestnet()
	client.SetOperator(accountId, accountKey)

	// Read file from disk
	localFileContents, err := os.ReadFile("my-file.txt")
	if err != nil {
		log.Fatalf("Error reading input file: %v", err)
	}

	// Write file onto Hedera Testnet, using HFS FileCreateTransaction
	fileCreateTx, err := hedera.NewFileCreateTransaction().
		// NOTE: File create transaction
		// Step (1) in the accompanying tutorial
		// /* ... */
		SetContents(localFileContents).
		FreezeWith(client)
	if err != nil {
		log.Fatalf("Error creating FileCreateTransaction: %v", err)
	}
	fileCreateTxSigned := fileCreateTx.Sign(accountKey)
	fileCreateTxSubmitted, err := fileCreateTxSigned.Execute(client)
	if err != nil {
		log.Fatalf("Error executing FileCreateTransaction: %v", err)
	}
	fileCreateTxId := fileCreateTxSubmitted.TransactionID
	fileCreateTxReceipt, err := fileCreateTxSubmitted.
		SetValidateStatus(true).
		GetReceipt(client)
	if err != nil {
		log.Fatalf("Error getting receipt for FileCreateTransaction: %v", err)
	}
	fileId := fileCreateTxReceipt.FileID

	// Read file from Hedera Testnet, using HFS FileContentsQuery
	fileReadQuery := hedera.NewFileContentsQuery().
		// NOTE: File contents query
		// Step (2) in the accompanying tutorial
		// /* ... */;
		SetFileID(*fileId)
	networkFileContents, err := fileReadQuery.Execute(client)
	if err != nil {
		log.Fatalf("Error executing for FileContentsQuery: %v", err)
	}

	txExplorerUrl := fmt.Sprintf("https://hashscan.io/testnet/transaction/%v", fileCreateTxId)
	fmt.Printf("fileId: %v\n", fileId)
	fmt.Printf("fileCreateTxId: %v\n", fileCreateTxId)
	fmt.Printf("txExplorerUrl: %v\n", txExplorerUrl)
	fmt.Printf("localFileContents: %s\n", localFileContents)
	fmt.Printf("networkFileContents: %s\n", networkFileContents)

	client.Close()
}
